test/parallel/test-async-hooks-http-parser-destroy.js

```diff
@@ -21,7 +21,12 @@ async_hooks.createHook({
     }
   }, N),
   destroy: (asyncId) => {
-    destroyedIds.push(asyncId);
+    if (createdIds.includes(asyncId)) {
+      // Work around https://github.com/nodejs/node/issues/26961.
+      // Remove the next line when #26961 is fixed.
+      if (!destroyedIds.includes(asyncId))
```

Comment:
I would rather not include this part and instead mark this test as flaky. Otherwise we might forget about it and the test will silently work while it should fail.

The failure indicates that we have a bug somewhere that we should fix.

Reply:
I see what you're saying and I'm definitely sympathetic. The problem is that with the changes in this PR, the test will basically fail 100% of the time. And I'm generally not OK with marking a test "flaky" as a workaround for a test that fails on every run. But the change is (IMO) a genuine improvement. It makes the checks more thorough (which is why it increases the failure rate dramatically).

I guess the solution everyone can support would be to fix the underlying bug.

Reply:
Now that https://github.com/nodejs/node/commit/ece507394a563bd7e3555a840a3d6980ffbb01ac landed, let's see if that fixes the remaining issues here and if I can remove this workaround.... Going to run CI with this and then again without it....

Reply:
Looks like `HTTPPARSER` ids never show up now.  I imagine that's unintended....

Reply:
Oh, it totally was intentional. Guess I need to update the test to use HTTPINCOMINGMESSAGE or something like that. Or maybe the test can/should be removed entirely?

Reply:
Probably the thing to do is to simply not filter the asyncIDs at all. If V8 compiling ever finishes for me locally, I'll try that out....

Reply:
Alas, it did not appear to fix the bug. IDs show up in `destroy` without showing up in `create`. But I should at least update the test to use HTTPINCOMINGMESSAGE.